deduplicate configuration and ensure it works on enumerate_camera_devices

Lunarequest opened the pull request:
## Linux
enumerate_camera_devices previously did not check if a camera configuration worked. This resulted in users being presented with a set of configurations, some of which did not work. enumerate_camera_devices now attempted to capture a frame. If this does not work the configuration is not returned. 

This greatly reduces the number of "false"/non working configurations. However there are some configurations which do not track but are still appended, I am unsure how to test these without requiring the camera be pointed at a person on startup.

before:
<img width="397" height="254" alt="image" src="https://github.com/user-attachments/assets/b3ce39f5-6843-4581-8db2-a614e79e8362" />

after:
<img width="396" height="153" alt="image" src="https://github.com/user-attachments/assets/f8f15048-da96-4fd1-839b-0f98dd9a88aa" />

## windows

On windows CAP_ANY returns 2 "entries" for every camera. Direct Show(DSHOW) and Media Foundation (MSMF) MSMF was introduced in windows vista and DSHOW has since been deprecated and its docs carry the warning it maybe ripped out in a future windows version. For future readiness only query by CAP_MSMF since that provides windows support

Lunarequest committed:
Ensure camera configuration works on enumerate_camera_devices

enumerate_camera_devices previously did not check if a camera
configuration worked. This resulted in users being presented with a set
of configurations, some of which did not work. enumerate_camera_devices
now attempted to capture a frame. If this does not work the
configuration is not returned. This greatly reduces the number of
"false" configurations. However there are some configurations which do
not track but are still appended, I am unsure how to test these without
requiring the camera be pointed at a person on startup.

Lunarequest committed:
Check if device is busy

If a device is busy(i.e) used by another device assume it works, to
tests this we check if we can open it using linuxpy and assume the
configuration works if we get a errno = 16 which is "Device or resource
busy"

Lunarequest committed:
Mediapipe: deduplicate cameras on windows

On windows CAP_ANY returns 2 "entries" for every camera. Direct Show
(DSHOW) and Media Foundation (MSMF) MSMF was introduced in windows vista
and DSHOW has since been deprecated and its docs carry the warning it
maybe ripped out in a future windows version. For future readiness only
filter by CAP_MSMF.

Drikanis commented:
# Windows testing

## Before PR

<img width="633" height="293" alt="image" src="https://github.com/user-attachments/assets/58065b5c-42a0-40b1-9579-3f5852578f2d" />

CAP_ANY:700.0 is the working camera on my system, starts within seconds of selecting it, and doesn't drop any packets.

## After PR:
First thing I noticed is that I get a version warning on loading the project, since SnekStudio is still on 4.4 but the project in the PR was last modified using 4.5. Not sure how much that matters though.

<img width="642" height="276" alt="image" src="https://github.com/user-attachments/assets/7b8a299e-5e20-45b6-bcc0-f23af3408184" />

The virtual cams are missing, which should technically still work but aren't my main camera, so might not be an issue for most people?

Selecting the only camera available (MSMF:0.0) takes a really long time to start tracking, but it eventually starts after a few minutes. It's also constantly dropping packets whereas the main version does not.

Lunarequest committed:
MediaPipe: fix performance issues on windows

We now set OPENCV_VIDEOIO_MSMF_ENABLE_HW_TRANSFORMS to "0" to fix the
performance issues based on https://github.com/opencv/opencv/issues/17687\
This for reasons beyond my understanding fixes the issue.

Lunarequest commented:
After a little more research what I believe is happening is the following, without `OPENCV_VIDEOIO_MSMF_ENABLE_HW_TRANSFORMS` being set to `0` causes the MSMF back end to stop using hardware accelerated transforms. I believe for some reason the HW transforms is *slower* then the software unaccelerated version mean the FPS from the camera exceeds what the hardware decoder provides? at least this is my theory but I have no way to test

Drikanis commented:
Or CAP_ANY was slower than "direct" MSMF? If the dropped frames are benign (which it seems to be) then I guess they can be ignored.

Aside from the loss of DSHOW-exclusive devices and the version bump on the project file, it seems fine from a testing perspective.

Lunarequest commented:
@ExpiredPopsicle can you look at this when you're free

Lunarequest committed:
update platform_status.json

Lunarequest committed:
Add fallback for win32 to CAP_DSHOW

If CAP_MSMF fails to get any cameras we now enumerate cameras using
DSHOW preserving the ability to run snudio in wine(if anyone is actually
doing that)
